src/openrct2/paint/sprite/Paint.Sprite.cpp

```diff
@@ -111,6 +112,12 @@ void sprite_paint_setup(paint_session* session, const uint16_t x, const uint16_t
         {
             case SPRITE_IDENTIFIER_VEHICLE:
                 vehicle_paint(session, (Vehicle*)spr, image_direction);
+#ifdef __ENABLE_LIGHTFX__
+                if (lightfx_for_vehicles_is_available())
+                {
+                    lightfx_add_lights_magic_vehicle((Vehicle*)spr);
```

Comment:
Please use a named cast (`static_cast`, `dynamic_cast`, ...) instead of c-style cast. In this case looks like static would do

Reply:
static_cast<Vehicle*>(spr);

gives an "invalid type conversion" error

Reply:
I changed it to

`lightfx_add_lights_magic_vehicle(reinterpret_cast<Vehicle*>(const_cast<rct_sprite*>(spr)));`

hope that's right

Reply:
I think it's fine for now, thanks